benchmark/url/whatwg-url-parse.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+const common = require('../common.js');
+const { URL } = require('url');
+const fs = require('fs');
+const path = require('path');
```

Comment:
Shouldn't this benchmark exist only in the ada repository? Honestly, I don't see a value in having it in both projects.

Reply:
> Shouldn't this benchmark exist only in the ada repository? Honestly, I don't see a value in having it in both projects.

The question is, do we want to know how fast can we fail if the input is invalid, or how fast we can process a successful URL? We need to measure both, and this is why this PR is needed on top of the existing benchmarks that uses the inputs from  web platform tests with almost 90% invalid URLs (because it is focusing on spec compliance).

Reply:
Currently, Ada only takes 11% of a URL parsing flow. It will be much more impactful if we have this benchmark in here too.

<img width="1616" alt="screenshot_2023-03-03_at_1 18 39_pm" src="https://user-images.githubusercontent.com/1935246/222923650-89b243ed-94e7-4663-8fb3-198e6aeeeda0.png">

Reply:
> (unrelated) @anonrig could you tell, what are you using here to get this performance stack trace?

Apple Instruments app, happy to write a blog post about it.

Reply:
Thank you!

A blog would actually be amazing!

Reply:
> Currently, Ada only takes 11% of a URL parsing flow.

Ok, that is a good argument to keep this benchmark.